<launch>
  <!-- Complete system launch file for autonomous navigation -->
  
  <!-- Robot Base Controller -->
  <node pkg="neovio_autonomous_car" type="robot_base_controller.py" name="robot_base_controller" output="screen">
    <param name="left_enable_pin" value="22"/>
    <param name="left_forward_pin" value="17"/>
    <param name="left_backward_pin" value="27"/>
    <param name="right_enable_pin" value="25"/>
    <param name="right_forward_pin" value="23"/>
    <param name="right_backward_pin" value="24"/>
    <param name="wheel_base" value="0.26"/>
    <param name="wheel_radius" value="0.065"/>
    <param name="max_speed" value="0.8"/>
  </node>

  <!-- RPLIDAR A1 -->
  <node name="rplidarNode" pkg="rplidar_ros" type="rplidarNode" output="screen">
    <param name="serial_port" type="string" value="/dev/ttyUSB0"/>
    <param name="serial_baudrate" type="int" value="115200"/>
    <param name="frame_id" type="string" value="laser"/>
    <param name="inverted" type="bool" value="false"/>
    <param name="angle_compensate" type="bool" value="true"/>
  </node>

  <!-- Transform from base_link to laser -->
  <node pkg="tf" type="static_transform_publisher" name="base_to_laser" 
        args="0 0 0.1 0 0 0 base_link laser 100"/>

  <!-- SLAM using gmapping -->
  <include file="$(find neovio_autonomous_car)/launch/gmapping.launch"/>

  <!-- Navigation stack -->
  <include file="$(find neovio_autonomous_car)/launch/navigation.launch"/>

  <!-- RViz for visualization -->
  <arg name="use_rviz" default="false"/>
  <node pkg="rviz" type="rviz" name="rviz" 
        args="-d $(find neovio_autonomous_car)/rviz/navigation.rviz" 
        if="$(arg use_rviz)"/>
</launch>

